### ISSUE TITLE
user access to queues persists after being removed in DB

### ISSUE THREAD

**Issue by ryanday36** [flux-framework/flux-accounting#332]
If a user who has been added to a queue via `flux account edit-user --queue=_queue_name user_name_` is removed from that queue with `flux account edit-user --queue= _user_name_`, their access to the queue persists after running an `account-priority-update`. E.g. on the management node:
```
[root@fluke1:~]# flux account view-user day36
creation_time   mod_time        active          username        userid          bank            default_bank    shares          job_usage       fairshare       max_running_jobsmax_active_jobs max_nodes       queues          projects        default_project 
1648069813      1679958340      1               day36           36075           dev             lc              1               72530.383804999990.99375         100             7               5               debug           *               *               
1648069811      1679958340      1               day36           36075           guests          lc              1               72530.383804999990.025           1               7               5               debug           *               *               
1648069812      1679958340      1               day36           36075           lc              lc              1               526456.357985   0.26875         100             7               5               debug           *               *               

[root@fluke1:~]# flux account edit-user --queues= day36
[root@fluke1:~]# flux account view-user day36
creation_time   mod_time        active          username        userid          bank            default_bank    shares          job_usage       fairshare       max_running_jobsmax_active_jobs max_nodes       queues          projects        default_project 
1648069813      1679958403      1               day36           36075           dev             lc              1               72530.383804999990.99375         100             7               5                               *               *               
1648069811      1679958403      1               day36           36075           guests          lc              1               72530.383804999990.025           1               7               5                               *               *               
1648069812      1679958403      1               day36           36075           lc              lc              1               526456.357985   0.26875         100             7               5                               *               *               

[root@fluke1:~]# /bin/flux account-priority-update
[root@fluke1:~]# date
Mon Mar 27 16:15:59 PDT 2023
[root@fluke1:~]#
```
but I can still submit to debug:
```
[day36@fluke108:~]$ date && flux run -n1 --queue debug hostname
Mon Mar 27 16:18:09 PDT 2023
fluke96
[day36@fluke108:~]$
```

**Comment by cmoussa1** [comment]
OK, I had a chance to write a reproducer in my Docker environment and the bug is as I expected; the `queues` list for a user in the plugin does not get cleared when it is cleared in the flux-accounting database. So, above, the user's `queues` list is empty in the DB, as confirmed by the output in `flux account view-user`, but the user's `queues` list still contains `debug` in its `bank_info->queues` list, which is why jobs still successfully submit to the queue.

So, my first thought is the behavior of the plugin should probably be adjusted to just clear the user's `queues` list upon a flux-accounting update and be re-populated with whatever is contained in the user's `queues` list in the flux-accounting database.

**Comment by cmoussa1** [comment]
Closing, as this should be resolved by #334.
